flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/api/config/ExecutionConfigOptions.java

```diff
@@ -60,9 +60,10 @@
 	// ------------------------------------------------------------------------
 
 	@Documentation.TableOption(execMode = Documentation.ExecMode.STREAMING)
-	public static final ConfigOption<String> TABLE_EXEC_SOURCE_IDLE_TIMEOUT =
+	public static final ConfigOption<Duration> TABLE_EXEC_SOURCE_IDLE_TIMEOUT =
 		key("table.exec.source.idle-timeout")
-			.defaultValue("-1 ms")
+			.durationType()
+			.defaultValue(Duration.ofMillis(0))
 			.withDescription("When a source do not receive any elements for the timeout time, " +
 				"it will be marked as temporarily idle. This allows downstream " +
 				"tasks to advance their watermarks without the need to wait for " +
```

Comment:
The default value is 0ms now. It would be better to mention the meaning of default value, otherwise it's confusing. 

> Default value is 0, which means detecting source idleness is not enabled.